deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,265 @@
+      {{- $_ := set $configs $name ($bytes | toString | fromYaml) }}
+    {{- end }}
+    {{- range $name := keys $configs | sortAlpha }}
+      {{- $_ := merge $.Values (get $configs $name) }}
+    {{- end }}
+
+    {{- $_ := set $.Values "defaults" (index $.Values.releases "$defaults") }}
+
+    {{- $_ := include "_stacks.update" . }}
+    {{- $_ := include "_releases.update" . }}
+
+    {{- range $condition, $message := $.Values.unsupported }}
+      {{- if eq "true" (include "_config.condition" (list $ $condition)) }}
+        {{- fail $message }}
+      {{- end }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config.lookup (list $ $path)
++-----------------------------------------------------------------------------------------
+| Look up a config setting in $.Values at $path.
+|
+| $path can be a '.' separated string of property names 'features.eirini.enabled', a
+| list of names ("stacks" $stacks "releases"), or a combination of both. The separator
+| '/' may also be used in place of the '.' for a more JSON-patch look.
+|
+| See comments on the _config._lookup implementation function for more details.
+|
+| This function calls _config.load to make sure the config object is initialized.
+==========================================================================================
+*/}}
+{{- define "_config.lookup" }}
+  {{- $root := first . }}
+  {{- $_ := include "_config.load" $root }}
+  {{- $query := join "." (rest .) | replace "/" "." }}
+  {{- include "_config._lookup" (concat (list $root.kubecf $root.Values) (splitList "." $query)) }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config.lookupManifest (list $ $path)
++-----------------------------------------------------------------------------------------
+| The same function as _config.lookup, except it searches $.kubecf.manifest instead.
+==========================================================================================
+*/}}
+{{- define "_config.lookupManifest" }}
+  {{- $_ := include "_config.load" (first .) }}
+  {{- $kubecf := get (first .) "kubecf" }}
+  {{- $query := join "." (rest .) | replace "/" "." }}
+  {{- include "_config._lookup" (concat (list $kubecf $kubecf.manifest) (splitList "." $query)) }}
+{{- end }}
+
+{{- /*
+==========================================================================================
+| _config._lookup (list $.kubecf $context $path)
++-----------------------------------------------------------------------------------------
+| Internal implementation of "_config.lookup" and "_config.lookupManifest".
+|
+| Lookup $path under $context and return either the value found, or the empty string.
+| Maps and slices are returned in JSON format; nil is returned as the empty string.
+|
+| $context is either an object (or an array). It normally starts out as $.Values
```

Comment:
```suggestion
| $context is either an object or an array. It normally starts out as $.Values
```

_x is either_ should be followed by _option 1 **or** option 2_, or something along those lines; so the alternative shouldn't be a parenthesized sub-clause.  Alternatively, we can do it this way (but I feel it's inferior):

```suggestion
| $context is an object (or an array). It normally starts out as $.Values
```

Reply:
Fixed in f3be1c0